pkg/storage/utils/decomposedfs/revisions.go

```diff
@@ -191,23 +207,65 @@ func (fs *Decomposedfs) RestoreRevision(ctx context.Context, ref *provider.Refer
 	nodePath := fs.lu.InternalPath(spaceID, kp[0])
 	var fi os.FileInfo
 	if fi, err = os.Stat(nodePath); err == nil {
-		// versions are stored alongside the actual file, so a rename can be efficient and does not cross storage / partition boundaries
-		versionsPath := fs.lu.InternalPath(spaceID, kp[0]+node.RevisionIDDelimiter+fi.ModTime().UTC().Format(time.RFC3339Nano))
+		// revisions are stored alongside the actual file, so a rename can be efficient and does not cross storage / partition boundaries
+		newRevisionPath := fs.lu.InternalPath(spaceID, kp[0]+node.RevisionIDDelimiter+fi.ModTime().UTC().Format(time.RFC3339Nano))
+
+		// touch new revision
+		if file, err := os.Create(newRevisionPath); err != nil {
+			return err
+		} else if err := file.Close(); err != nil {
+			return err
+		}
```

Comment:
At any of the error returns following here, a new revision file in inconsistent stage is left behind. Shouldn't we have a `defer` that removes the file in case of error and closes the descriptor?

Reply:
agreed